From 9cbdf3a6e2cb23a7673fe19b2a3f888714095019 Mon Sep 17 00:00:00 2001
From: Oliver Kurth <okurth@vmware.com>
Date: Wed, 1 Apr 2020 11:31:36 -0700
Subject: [PATCH] Add a Coverity annotation.

The Coverity scan of open-vm-tools reports an out-of-bounds access
issue in Util_BacktraceWithFunc.  This is intentional, to compute
the base pointer for a stack trace.
---
 lib/user/utilBacktrace.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/user/utilBacktrace.c b/lib/user/utilBacktrace.c
index dca33d267..e1259606f 100644
--- a/lib/user/utilBacktrace.c
+++ b/lib/user/utilBacktrace.c
@@ -1,5 +1,5 @@
 /*********************************************************
- * Copyright (C) 2013-2019 VMware, Inc. All rights reserved.
+ * Copyright (C) 2013-2020 VMware, Inc. All rights reserved.
  *
  * This program is free software; you can redistribute it and/or modify it
  * under the terms of the GNU Lesser General Public License as published
@@ -438,6 +438,7 @@ Util_BacktraceWithFunc(int bugNr,                // IN:
    } else {
       outFunc(outFuncData, "Backtrace for bugNr=%d\n",bugNr);
    }
+   /* coverity[ptr_arith] */
    UtilBacktraceFromPointerWithFunc(&x[-2], outFunc, outFuncData);
 #endif
 }
